proto/exocore/operator/v1/tx.proto

```diff
@@ -0,0 +1,166 @@
+import "cosmos/msg/v1/msg.proto";
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/ExocoreNetwork/exocore/x/operator/types";
+
+// DecValueField is a field that holds a value of sdk.LegacyDec type.
+message DecValueField {
+  // amount is the USD value of the asset, as an sdk.LegacyDec.
+  string amount = 1
+  [
+    (cosmos_proto.scalar) = "cosmos.Dec",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+}
+
+// ClientChainEarningAddrList is the list of client chain earning addresses.
+message ClientChainEarningAddrList {
+  // earning_info_list is the contents of ClientChainEarningAddrList.
+  repeated ClientChainEarningAddrInfo earning_info_list = 1;
+}
+
+// ClientChainEarningAddrInfo is the client chain earning address info.
+message ClientChainEarningAddrInfo {
+  // lz_client_chain_id is the layer0 client chain id.
+  uint64 lz_client_chain_id = 1  [(gogoproto.customname) = "LzClientChainID"];
+  // client_chain_earning_addr is the client chain earning address.
+  string client_chain_earning_addr = 2;
+}
+
+// OperatorInfo is the operator info.
+message OperatorInfo {
+  // earnings_addr is the earnings address.
+  string earnings_addr = 1;
+  // approve_addr is the approve address.
+  string approve_addr = 2;
+  // operator_meta_info is the operator meta info.
+  string operator_meta_info = 3;
+  // client_chain_earning_addr_list is the client chain earning address list.
+  ClientChainEarningAddrList client_chain_earnings_addr = 4;
+}
+
+// OptedInfo is the opted information about operator
+message OptedInfo {
+  // slash_contract is the slash contract address of AVS opted-in by the operator
+  string slash_contract = 1;
+  // opted_in_height is the height at which the operator opted in
+  uint64 opted_in_height = 2;
+  // opted_out_height is the height at which the operator opted out
+  uint64 opted_out_height = 3;
+}
+
+// OptedInAssetState is the state of opted-in asset
+message OptedInAssetState{
+  // amount of the opted-in asset
+  string amount = 1
+  [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable) = false
+  ];
+  // value is the USD value of the opted-in asset
+  string value = 2
+  [
+    (cosmos_proto.scalar) = "cosmos.Dec",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+}
+
+// OperatorSlashInfo is the slash info of operator
+message OperatorSlashInfo {
+  // slash_contract is the address of slash contract
+  string slash_contract = 1;
+  // slash_height is the height at which the slash event is submitted
+  int64 slash_height = 2;
+  // occurred_height is the height at which the slash event occurs
+  int64 occurred_height = 3;
+  // execute_height is the height at which the slash event is executed
+  int64 execute_height = 4;
```

Comment:
why the slash_height and execute_height is different? In that case, better name like this
submitted_height
processed_height
event_height

Reply:
Yeah, I will rename them and add some comments.